electron/js/menu/TrayIconHandler.js

```diff
@@ -0,0 +1,85 @@
+const {app, Menu, nativeImage, Tray} = require('electron');
+const config = require('./../config');
+const environment = require('./../environment');
+const lifecycle = require('./../lifecycle');
+const locale = require('./../../locale/locale');
+const path = require('path');
+const windowManager = require('./../window-manager');
+
+class TrayIconHandler {
+  constructor(appIcon = new Tray(nativeImage.createEmpty())) {
+    this.appIcon = appIcon;
+    this.icons = this.createIcons();
+    this.lastUnreadCount = 0;
+    this.initTrayIcon();
+  }
+
+  initTrayIcon() {
+    this.appIcon.setImage(this.icons.tray);
+    if (this.hasTrayMenuSupport) {
+      this.buildTrayMenu();
+    }
+  }
+
+  get defaultImageExtension() {
+    return environment.platform.IS_WINDOWS ? 'ico' : 'png';
+  }
+
+  get hasTrayMenuSupport() {
+    return !environment.platform.IS_MAC_OS;
+  }
+
+  get hasOverlaySupport() {
+    return environment.platform.IS_WINDOWS;
+  }
+
+  buildTrayMenu() {
```

Comment:
No need to expose this to the public

Reply:
The case of this one is a bit trickier because it's a bit bigger. 
So, in order not to bloat the public API of this class, a few solutions include:

- still inlining it in the `constructor` ;
- define it as a external function and call it with `call` (`buildTrayMenu.call(this)`) ;
- use the `_` convention to prevent misuse of the method. 

I think I would for the the `call` solution but it's up for discussion of course.